paasta_tools/deployd/watchers.py

```diff
@@ -225,20 +229,20 @@ def process_default(self, event):
             for service_instance in service_instances:
                 self.filewatcher.inbox_q.put(service_instance)
 
-    def get_service_instances_with_changed_id(self):
-        marathon_app_ids = [app.id.lstrip('/') for app in get_marathon_apps()]
-        instances = get_services_for_cluster(cluster=self.public_config.get_cluster(),
-                                             instance_type='marathon',
-                                             soa_dir=DEFAULT_SOA_DIR)
-        service_instances = []
-        for service, instance in instances:
-            config = load_marathon_service_config(service=service,
-                                                  instance=instance,
-                                                  cluster=self.public_config.get_cluster(),
-                                                  soa_dir=DEFAULT_SOA_DIR)
-            if config.format_marathon_app_dict()['id'] not in marathon_app_ids:
-                service_instances.append((service, instance))
-        return service_instances
+
+def get_service_instances_with_changed_id(instances, cluster):
+    marathon_app_ids = [app.id.lstrip('/') for app in get_marathon_apps()]
+    print(marathon_app_ids)
+    service_instances = []
+    for service, instance in instances:
+        config = load_marathon_service_config(service=service,
+                                              instance=instance,
+                                              cluster=cluster,
+                                              soa_dir=DEFAULT_SOA_DIR)
```

Comment:
Yeah, this is just laziness really. I think it should probably be a config variable rather than a cli variable... I'm going to leave this be because it's hardcoded in a few places but happy to fix it in a separate PR.